File: core-api/src/main/java/com/base/coreapi/model/oscc/dto/StructureUpdateDTO.java
package com.base.coreapi.model.oscc.dto;

import com.base.coreapi.model.oscc.Version;
import com.base.coreapi.model.oscc.OsccObject;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class StructureUpdateDTO {

    private Version version;

    private List<OsccObject> osccObjects;
}

File: frontend-app/src/app/model/structure-update-dto.ts
import {VersionOfType} from "./version-of-type.model";
import {OsccObject} from "./object-model";

export class StructureUpdateDto {

  public versionOfTypes: VersionOfType[];

  public objects: OsccObject[];

  constructor(objects: OsccObject[], versionOfTypes: VersionOfType[]){
    this.objects = objects;
    this.versionOfTypes = versionOfTypes;
  }
}
